src/fifobuffer.jl

```diff
@@ -79,10 +80,10 @@ Base.read(f::FIFOBuffer) = readavailable(f)
 Base.flush(f::FIFOBuffer) = nothing
 
 function Base.eof(f::FIFOBuffer)
-    return f.eof && f.nb == 0
+    return f.content_length >= 0 && f.content_pos >= f.content_length || f.content_length < 0 && f.nb == 0
 end
 function Base.close(f::FIFOBuffer)
-    f.eof = true
+    f.content_length = f.content_pos
```

Comment:
I guess it somewhat depends on if `content_pos` is tracking the # of bytes *written* to the buffer, or # of bytes *read* from buffer. I think the latter is simpler and is probably why you went that route?